File: 백준/Gold/22115. 창영이와 커피/창영이와 커피.js
let input = require("fs").readFileSync('/dev/stdin').toString();
input = input.split("\n")
    
const [n, k] = input[0].split(" ").map(Number)
const dp = new Array(k+1).fill(1e9)
dp[0]=0

input = input[1].split(" ").map(Number)
for (let i = 0; i < n; i++) {
    let now = input[i]
    for (let j = k; j>=now; j--) {
        if(dp[j-now] === 1e9) continue
        dp[j]=Math.min(dp[j],dp[j-now]+1)
    }
    if (now<=k){
        dp[now]=1
    }
}

console.log(dp[k]===1e9?-1:dp[k])

File: 백준/Gold/22115. 창영이와 커피/창영이와 커피.py
import sys;input=sys.stdin.readline

n, k = map(int, input().split())
dp=[101]*(100001)
dp[0]=0

coffees = list(map(int, input().split()))
for i in range(n):
    t=coffees[i]
    for j in range(k,t-1,-1):
        dp[j]=min(dp[j],dp[j-t]+1)
    dp[t]=1

print(-1 if dp[k]==101 else dp[k])